### PR TITLE
Line Highlighting

### PR DESCRIPTION
**Pull request by pierrebeitz**
@russurquhart1 recently asked whether we could add asciidoctor to our stack. it turned out that the most-wanted feature was line-highlighting.

Job Examples - D2iQ Docs 2020-08-24 11-42-16

to have a look or play around with that yourself:

the `data-line` attribute can take the following values:

* `1` - highlight line 1.
* `1-3` - highlight the range 1 to 3.
* `1,3` - highlight the lines 1 and 3.
* `1-3,42` - highlight lines 1 to 3 and 42.

### CODE REVIEW

File: pages/mesosphere/dcos/2.1/deploying-jobs/examples/index.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ These examples provide common usage scenarios for jobs.
 This JSON file creates a simple job with no schedule.
 
 1. Create a JSON file with the following contents.
-    ```json
+    ```json {data-line=2-5}
```
Comment (pierrebeitz) [inline pages/mesosphere/dcos/2.1/deploying-jobs/examples/index.md:26]: @russurquhart1 @cmastr this is how you'd highlight lines. the rest of this PR is enablement.

**Comment by cmastr** [comment]
I can take a look and run this on Wednesday if that's ok.

File: js/code-container.js [hunk 2]
```diff
@@ -1,8 +1,315 @@
...
     $(elem).wrapAll('<div class="code-container" />');
+    $(elem).attr("data-line", $(elem).find("code").attr("data-line"));
+    $(elem).attr("data-range", $(elem).find("code").attr("data-range"));
     elem.classList.add("language-*");
   });
 }
 
 wrapCodeBlocks();
+
+(function () {
```
Comment (pierrebeitz) [inline js/code-container.js:12]: copied from here: https://prismjs.com/plugins/line-highlight/

File: scss/vendor/_prism.scss [hunk 3]
```diff
@@ -179,3 +179,66 @@
 	 color: inherit;
 	 text-decoration: none;
  }
+
+//////////////////////////////////////////////////////////////////////////////
+//                          LINE HIGHLIGHTING PLUGIN                        //
+//////////////////////////////////////////////////////////////////////////////
+pre[data-line] {
```
Comment (pierrebeitz) [inline scss/vendor/_prism.scss:186]: copied from here with slight adjustments: https://prismjs.com/plugins/line-highlight/

**Comment by samvantran** [comment]
Nice 👍 lgtm

**Comment by cmastr** [comment]
Looks good to me too!

**Comment by cmastr** [comment]
Need to add this to the wiki.

**Comment by pierrebeitz** [comment]
i'd say we should not merge without changing the CSS a bit. shall we let Daniel Schwartz have a look? 
or do we just remove those numbers to the left and are happy for a first iteration?

**Comment by cmastr** [comment]
> i'd say we should not merge without changing the CSS a bit. shall we let Daniel Schwartz have a look?
> or do we just remove those numbers to the left and are happy for a first iteration?

Let me ping him and see what he thinks.

**Comment by danielone** [comment]
@mperrotti and Nan working on code viewer for dispatch (and ui kit) that displays line number very nicely in more traditional format (solid column) From here,  can then just highlight a diff color/tint the rows that need to be highlighted. The circle style feels a bit jarring with the round geometry. 

https://mesosphere.invisionapp.com/share/G2YHZ37NK4T#/424355157_Failed_Logs_2
pass: pdux@d2iq2020!

The fade out is pretty cool =)

**Comment by pierrebeitz** [comment]
Added an according section to the README. I think we could merge now.

**Pull request merged**